Documentation/coding-guidelines/project-guidelines.md

```diff
@@ -1,50 +1,84 @@
-#Library Project guidelines
-Library projects should use the following directory layout.
-```
-src\<Library Name>\src - Contains the source code for the library.
-src\<Library Name>\ref - Contains any reference assembly projects for the library
-src\<Library Name>\pkg - Contains package projects for the library.
-src\<Library Name>\tests - Contains the test code for a library
-```
-In the src directory for a library there should be only **one** `.csproj` file that contains any information necessary to build the library in various configurations (see [Configurations](#project-configuration-conventions)). The src directory should also contain exactly **one** `.builds` file which contains all the valid configurations that a library should be built for (see [#.builds file](#project-builds-file)).
+#Build Project Guidelines
+In order to work in corefx repo you must first run build.cmd/sh from the root of the repo at least
+once before you can iterate and work on a given library project.
+
+##Behind the scenes with build.cmd
+
+- Setup tools (currently done in init-tools but will later be a boot-strap script in run.cmd/sh)
+- Restore external dependencies
+ - CoreCLR - Copy to bin\runtime\[BuildTarget]
+ - Netstandard Library - Copy to bin\ref\netstandard
+ - UAP - Copy to bin\runtime\uap??
+- Build targeting pack
+ - Build src\ref.builds file which will build all the references assemblies to bin\ref\[BuildTargetFramework]. Where BuildTargetFramework is a versionless TFM representing the
```

Comment:
I'm not sure I understand why would the TFM be versionless. Won't we x-compile for older versions of TFMs? Will everything built out of the repo be `V-Next`?